services/QuillLMS/app/views/application/_google_integration_change_banner.html.erb

```diff
@@ -0,0 +1,46 @@
+<% current_path = request.env['PATH_INFO'] %>
+<% banner_link = "https://support.quill.org/en/articles/8496308-how-to-sign-into-a-google-linked-quill-account-with-a-quill-password" %>
+
+<% if Date.today < Date.new(2023, 12, 23) && ['session/new', 'sign-up/student', 'sign-up/teacher'].any? { |str| current_path.include?(str) } %>
+  <div class="banner google-integration-change-banner" id="google-login-and-signup-banner">
+    <div class="content-container">
+      <p>Having trouble signing in with Google? Create a password for Quill or request app verification. <a href="<%= banner_link %>" rel="noopener noreferrer" target="_blank">Learn more</a></p>
+    </div>
+  </div>
+<% elsif current_user&.teacher? && current_user&.google_id %>
+  <div class="banner google-integration-change-banner" id="google-teacher-banner">
+    <div class="content-container">
+      <p>Google Classroom teachers, if your students are having trouble signing in, ask them to create a Quill password or request app verification. <a href="<%= banner_link %>" rel="noopener noreferrer" target="_blank">Learn more</a></p>
+    </div>
+    <button aria-label="Dismiss the banner" class="interactive-wrapper" id="close-google-teacher-banner"><img alt="" src="<%= "#{ENV['CDN_URL']}/images/icons/close-white.svg" %>"></button>
+  </div>
+<% end %>
+
+<script>
+  document.addEventListener('DOMContentLoaded', function() {
+    const bannerId = 'google-teacher-banner'
+    const cookieName = 'google_teacher_banner_closed'
+
+    const bannerElement = document.getElementById(bannerId);
+    const closeButton = document.getElementById(`close-${bannerId}`);
+
+    if (bannerElement && closeButton) {
+      if (document.cookie.indexOf(`${cookieName}=1`) === -1) {
+        bannerElement.style.display = 'block';
+      }
+
+      closeButton.addEventListener('click', function(e) {
+        e.preventDefault();
+        bannerElement.style.display = 'none';
+        document.cookie = `${cookieName}=1; path=/`;
+      });
+    }
+
+    const loginSignupBanner = document.getElementById('google-login-and-signup-banner');
+
+    if (document.cookie.indexOf('google_login_and_signup_banner_closed=1') === -1 && loginSignupBanner) {
+      loginSignupBanner.style.display = 'block';
+    }
+
+  });
+</script>
```

Comment:
If this banner is non-dismissable, do we even need this `script` block?

Reply:
@anathomical the one that shows up on the sign up and log in pages (just through December) is not dismissable, but the one that shows up for logged-in teachers is.